data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/AbstractGetSimpleDataFromState.java

```diff
@@ -39,6 +40,10 @@ public AbstractGetSimpleDataFromState(
   public void handleRequest(RestApiRequest request) throws JsonProcessingException {
     final SafeFuture<Optional<StateAndMetaData>> future =
         chainDataProvider.getBeaconStateAndMetadata(request.getPathParameter(PARAMETER_STATE_ID));
+    if (future == null) {
+      request.respondError(SC_BAD_REQUEST, "Bad Request - check state_id is correct");
+      return;
+    }
     request.respondAsync(
         future.thenApply(
```

Comment:
i wonder if i put bad request on the end of this future processing as an ExceptionallyCompose, but then I guess that's potentially going to hide issues...